src/h_vialib/secure/url.py

```diff
@@ -0,0 +1,108 @@
+"""Routines for signing and checking URLs."""
+
+from datetime import timedelta
+from urllib.parse import parse_qs, parse_qsl, urlencode, urlparse
+
+from h_vialib.exceptions import InvalidToken
+from h_vialib.secure import SecureToken, quantized_expiry
+
+
+class SecureURL(SecureToken):
+    """Sign and check URLs with a JWT."""
+
+    def __init__(self, secret, token_param):
+        """Initialise the SecureURL.
+
+        :param secret: Secret to sign and check with
+        :param token_param: The URL parameter to use for the token
+        """
+        super().__init__(secret)
+        self._token_param = token_param
+
+    def create(
+        self, url, payload, expires=None, max_age=None
+    ):  # pylint: disable=arguments-differ
+        """Create a signed URL which can be checked with this class.
+
+        The entire URL should be added that you want to sign, without any
+        modifications. This will remove any existing signature, create a new
+        one, and append it to the URL as the parameter
+
+        :param url: The URL to sign
+        :param payload: Dict of extra information to put in the token
+        :param expires: Datetime by which this token with expire
+        :param max_age: ... or max age in seconds after which this will expire
+        :return: A URL with an extra parameter
+
+        :raise ValueError: if neither expires nor max_age is specified, or no
+            URL is provided
+        """
+        if not url:
+            raise ValueError("A URL is required to create a token")
+
+        url = self._strip_token(url)
```

Comment:
Should we be truncating the URL in anyway? The maximum length of a URL is going to be about 4000 chars. If we take a URL which is 2000 chars, our token will definitely be longer.